meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -1,183 +1,213 @@
-#                                                                                                                                                              4MiB
-
-# check if we are running on a poky version which deploys to IMGDEPLOYDIR instead of DEPLOY_DIR_IMAGE (poky morty introduced this change)
+RESIN_ROOT_FSTYPE ?= "ext4"
+
 python() {
+    # Check if we are running on a poky version which deploys to IMGDEPLOYDIR
+    # instead of DEPLOY_DIR_IMAGE (poky morty introduced this change)
     if d.getVar('IMGDEPLOYDIR', True):
-        d.setVar('RESIN_SDIMG_ROOTFS', '${IMGDEPLOYDIR}/${IMAGE_NAME}.rootfs.${RESIN_SDIMG_ROOTFS_TYPE}')
-        d.setVar('RESIN_SDIMG', '${IMGDEPLOYDIR}/${IMAGE_NAME}.rootfs.resin-sdcard')
+        d.setVar('RESIN_ROOT_FS', '${IMGDEPLOYDIR}/${IMAGE_NAME}.rootfs.${RESIN_ROOT_FSTYPE}')
+        d.setVar('RESIN_RAW_IMG', '${IMGDEPLOYDIR}/${IMAGE_NAME}.rootfs.resinos-img')
     else:
-        d.setVar('RESIN_SDIMG_ROOTFS', '${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfs.${RESIN_SDIMG_ROOTFS_TYPE}')
-        d.setVar('RESIN_SDIMG', '${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfs.resin-sdcard')
+        d.setVar('RESIN_ROOT_FS', '${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfs.${RESIN_ROOT_FSTYPE}')
+        d.setVar('RESIN_RAW_IMG', '${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfs.resinos-img')
 }
 
-# This image depends on the rootfs image
-IMAGE_TYPEDEP_resin-sdcard = "${RESIN_SDIMG_ROOTFS_TYPE}"
 
-# Partition labels
 RESIN_BOOT_FS_LABEL ?= "resin-boot"
-RESIN_ROOT_FS_LABEL ?= "resin-root"
-RESIN_UPDATE_FS_LABEL ?= "resin-updt"
-RESIN_CONFIG_FS_LABEL ?= "resin-conf"
+RESIN_ROOTA_FS_LABEL ?= "resin-rootA"
+RESIN_ROOTB_FS_LABEL ?= "resin-rootB"
+RESIN_STATE_FS_LABEL ?= "resin-state"
 RESIN_DATA_FS_LABEL ?= "resin-data"
 
-# Used as extension for fingerprint files
-FINGERPRINT_EXT ?= "fingerprint"
-
-# Boot partition size [in KiB] (will be rounded up to IMAGE_ROOTFS_ALIGNMENT)
-RESIN_BOOT_SPACE ?= "40960"
-
-# Set alignment to 4MB [in KiB]
-IMAGE_ROOTFS_ALIGNMENT = "4096"
+# Sizes in KiB
+RESIN_BOOT_SIZE ?= "40960"
+RESIN_ROOTB_SIZE ?= ""
+RESIN_STATE_SIZE ?= "20480"
+RESIN_IMAGE_ALIGNMENT ?= "4096"
+IMAGE_ROOTFS_ALIGNMENT = "${RESIN_IMAGE_ALIGNMENT}"
 
-RESIN_SDIMG_ROOTFS_TYPE ?= "ext4"
-
-# Default bootloader to virtual/bootloader
+RESIN_FINGERPRINT_EXT ?= "fingerprint"
+RESIN_FINGERPRINT_FILENAME ?= "resinos"
+RESIN_BOOT_FINGERPRINT_PATH ?= "${WORKDIR}/${RESIN_FINGERPRINT_FILENAME}.${RESIN_FINGERPRINT_EXT}"
 RESIN_IMAGE_BOOTLOADER ?= "virtual/bootloader"
+RESIN_IMAGEDATESTAMP = "${@time.strftime('%Y.%m.%d',time.gmtime())}"
+RESIN_RAW_IMG_COMPRESSION ?= ""
+RESIN_DATA_FS = "${DEPLOY_DIR}/images/${MACHINE}/${RESIN_DATA_FS_LABEL}.img"
+RESIN_BOOT_FS = "${WORKDIR}/${RESIN_BOOT_FS_LABEL}.img"
+RESIN_STATE_FS = "${WORKDIR}/${RESIN_STATE_FS_LABEL}.img"
+
+
+IMAGE_TYPEDEP_resinos-img = "${RESIN_ROOT_FSTYPE}"
```

Comment:
maybe keep the comment also for this above statement:

"# This image depends on the rootfs image"